### PR TITLE
Allow the registry conflict warning to be disabled

### PR DESCRIPTION
**Pull request by jlowin**
When developing interactively or building a framework that dynamically generates tasks, my terminal is full of warnings about name conflicts in the Prefect registry. In most situations this has no consequence, but the warnings pollute my output. This introduces a setting to disable the warning, but the default is to leave it enabled (current behavior).

### CODE REVIEW

**Review by zzstoatzz — APPROVED** [review]
🙌

File: src/prefect/flows.py [hunk 1]
```diff
@@ -347,18 +348,23 @@ def __init__(
...
-            for other in registry.get_instances(Flow)
-            if other.name == self.name and id(other.fn) != id(self.fn)
+        if (
+            registry
+            and any(
+                other
+                for other in registry.get_instances(Flow)
+                if other.name == self.name and id(other.fn) != id(self.fn)
+            )
+            and PREFECT_WARN_OBJECT_REGISTRY_CONFLICT.value()
```
Comment (desertaxle) [inline src/prefect/flows.py:358]: nit: The `.value` isn't needed.
```suggestion
            and PREFECT_WARN_OBJECT_REGISTRY_CONFLICT
```

File: src/prefect/tasks.py [hunk 2]
```diff
@@ -335,10 +336,14 @@ def __init__(
...
-            for other in registry.get_instances(Task)
-            if other.name == self.name and id(other.fn) != id(self.fn)
+        if (
+            registry
+            and any(
+                other
+                for other in registry.get_instances(Task)
+                if other.name == self.name and id(other.fn) != id(self.fn)
+            )
+            and PREFECT_WARN_OBJECT_REGISTRY_CONFLICT.value()
```
Comment (desertaxle) [inline src/prefect/tasks.py:346]:
```suggestion
            and PREFECT_WARN_OBJECT_REGISTRY_CONFLICT
```

**Review by desertaxle — APPROVED** [review]
LGTM!

**Review by serinamarie — APPROVED** [review]

**Pull request merged**